src/indexing.c

```diff
@@ -266,6 +266,36 @@ indexing_create_and_verify_hypertable_indexes(Hypertable *ht, bool create_defaul
 	table_close(tblrel, AccessShareLock);
 }
 
+bool
+ts_relation_has_primary_or_unique_index(Relation htrel)
+{
+	List *indexoidlist = RelationGetIndexList(htrel);
+	ListCell *lc;
+	bool result = false;
+
+	if (htrel->rd_pkindex != InvalidOid)
+		return true;
+
+	foreach (lc, indexoidlist)
+	{
+		Oid indexoid = lfirst_oid(lc);
+		HeapTuple indexTuple;
+		Form_pg_index index;
+
+		indexTuple = SearchSysCache1(INDEXRELID, ObjectIdGetDatum(indexoid));
+		if (!HeapTupleIsValid(indexTuple)) /* should not happen */
+			elog(ERROR, "cache lookup failed for index %u", indexoid);
+		index = (Form_pg_index) GETSTRUCT(indexTuple);
+		result = index->indisunique || index->indisprimary;
```

Comment:
i think you only need to check for indisunique here
```
dev=# select * from pg_index where indisprimary=true and indisunique!=true;
 indexrelid | indrelid | indnatts | indnkeyatts | indisunique | indisprimary | indisexclusion | indimmediate | indisclustered | indisvalid | indcheckxmin | indisready | indislive | indisreplident | indkey | indcollation | indclass | indo
ption | indexprs | indpred
------------+----------+----------+-------------+-------------+--------------+----------------+--------------+----------------+------------+--------------+------------+-----------+----------------+--------+--------------+----------+-----
------+----------+---------
(0 rows)

Time: 1.333 ms
```

Reply:
removed redundant check